app/forms/DeclarationPage.scala

```diff
@@ -16,4 +16,18 @@
 
 package forms
 
-trait DeclarationPage
+import models.DeclarationType.{CLEARANCE, DeclarationType}
+import models.viewmodels.TariffContentKey
+
+trait DeclarationPage {
+  def defineTariffContentKeys(decType: DeclarationType): Seq[TariffContentKey] = Seq.empty
+
+  def defineJourneySpecificKey(decType: DeclarationType): String = "common"
```

Comment:
I would like to challenge the presence of this method (and the `ClearanceModifierPage` class). This method is not used everywhere and therefore it makes me wonder if it is not introducing unnecessary complexity.
For e.g. in `ConsignmentReferences` there are still 'clearance' and 'common' being explicitly repeated in every key. On the other hand in `CusCode` there is only one possible key but it contains `common` suffix. Even though the default implementation of `defineJourneySpecificKey` would return the same suffix.

I'd suggest to remove this method if it is possible to have only a single one. From the PR it seems to me that it is not necessary. And actually, maybe it will be easier to read through those keys if they are not built from several methods?

Reply:
After our conversation I have removed the `ClearanceModifierPage` class as it really was not adding anything useful and moved its implementation of `defineJourneySpecificKey` (renamed for clarity to `getJourneyTypeSpecialisation`) to the base `DeclarationPage` trait.

I have not moved the match statement into every concretion of the `DeclarationPage` trait as it offended by DRY principles too much :smile: I tried to think of other ways to implement the logic but it all boils down to the fact we have 2 different concerns that are determined by declaration type: 1) The number of paragraphs in a tariff expander, 2) The content of each paragraph in the tariff expander.

The potential solutions to centralise these two concerns in one base trait caused far too much complexity and obfuscation of the code's purpose IMO. We can discuss again on Monday.